src/ai/storage.py

```diff
@@ -30,8 +30,8 @@
 # currently 1 minute
 RELEASE_INTERVAL_SECONDS = 60
 
-REJECT_MESSAGE = 'Invalid input format. Use `[DRONE ID HERE] :: [TARGET DRONE HERE] :: [INTEGER BETWEEN 1 - 24 HERE] :: [RECORDED PURPOSE OF STORAGE HERE]` (exclude brackets).'
-MESSAGE_FORMAT = r'^(\d{4}) :: (\d{4}) :: (\d+) :: (.*)'
+REJECT_MESSAGE = 'Invalid input format. Use `[DRONE ID HERE] :: [TARGET DRONE HERE] :: [NUMBER BETWEEN 1 - 24 HERE] :: [RECORDED PURPOSE OF STORAGE HERE]` (exclude brackets).'
+MESSAGE_FORMAT = r'^(\d{4}) :: (\d{4}) :: ((?:\d{0,2}\.\d{1,2})|(?:\d{1,2}\.?\d{0,2})) :: (.+)'
```

Comment:
Doesn't the regex also allow input of 0.5 which is less than 1? That's fine in my opinion but the reject message does not reflect that correctly.
Should maybe be changed to 
`[NUMBER BETWEEN 0 - 24 HERE]`

Reply:
I've updated it to say "UP TO 24", as "0 - 24" wouldn't be quite right either as 0 is not allowed, and "0.01 to 24" is a bit cumbersome.

I have changed the regex as well because I realised that I was being too pedantic with it: it's better to let any number through and let the 0-24 test reject it rather than checking the number of digits with the regex.  The error message it returns will be more helpful, and there's no need for an error message at all if you specify too many decimal places, it will just round them off.